### ISSUE TITLE
Support for Mac and Windows

### ISSUE THREAD

**Issue by GrbavaCigla** [mpromonet/libv4l2cpp#18]
Hi, 

I am writing a client for a streamer and it needs to be cross-platform. If you can't port it, would you merge my pull request?

Thanks in advance.

**Comment by mpromonet** [comment]
Hi Alexa,

I have no mac, but on windows I guess it will be a little tricky, v4l2 interface is video for Linux then Linux oriented. If you are looking for a cross-platform streamer, there are some solution like ffmpeg, vlc, gstreamer,.... 

My approach (not only on this project) is that the code length & complexity should be in relation with the feature it brings.   

Then if you submit a pull request that fix compilation on mac, I will probably merge, but if you submit a pull request that completly change the interface and multiply per 3 or 10 the length of the code, I will probably not merge it.   
On this project, there is pending PR fixing some clang warning, this is probably good, but I don't see the advantage to replace <errno.h> with <cerrno>, or memset fix {}.    
In conclusion, it depends ..... 

Best Regards,
Michel.
